src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -631,7 +631,8 @@ CHIP_ERROR DiscoveryImplPlatform::ResolveNodeId(const PeerId & peerId)
 {
     // Resolve requests can only be issued once DNSSD is initialized and there is
     // no caching currently
-    VerifyOrReturnError(mState == State::kInitialized, CHIP_ERROR_INCORRECT_STATE);
+    // When the DNSSD is kInitializing, can also try to resolve.
```

Comment:
Is this true on all platforms?   Does the API documentation clearly say that?

Because the comment right above the bit you are adding flatly contradicts what your new comment says.

What sort of audit was done here?  Why does the existing comment say what it does?

Reply:
I have updated the comment

Reply:
That answers literally none of the questions I asked.

Reply:
So for example: is it true that _on all platforms_ when the state is kInitializing it is ok to try to resolve?  Is there anything that _guarantees_ that this will be ok?  I see nothing that would guarantee that.